File: src/main/java/dev/wiisportsresorts/duelsplugin/DuelRequest.java
package dev.wiisportsresorts.duelsplugin;

import org.bukkit.entity.Player;

public class DuelRequest {
  public Player requester;
  public Player receiver;
  public DuelType type;

  public DuelRequest(DuelType type, Player requester, Player receiver) {
    this.type = type;
    this.requester = requester;
    this.receiver = receiver;
  }

  public boolean includesPlayer(Player player) {
    return includesPlayer(player.getName());
  }

  public boolean includesPlayer(String playerName) {
    return Duels.sumo.current().receiver.getName().equals(playerName) ||
      Duels.sumo.current().requester.getName().equals(playerName);
  }

  public Player not(Player player) {
    if (Duels.sumo.current().receiver.getName().equals(player.getName()))
      return requester;
    else if (Duels.sumo.current().requester.getName().equals(player.getName()))
      return receiver;

    return null;
  }
}

File: src/main/kotlin/dev/wiisportsresorts/duelsplugin/DuelRequest.kt
package dev.wiisportsresorts.duelsplugin

import org.bukkit.entity.Player

class DuelRequest(val type: DuelType, val requester: Player, val receiver: Player) {

    fun includesPlayer(player: Player): Boolean {
        return receiver.name == player.name || requester.name == player.name
    }

    fun not(player: Player): Player? {
        if (receiver.name == player.name) return requester
        else if (requester.name == player.name) return receiver

        return null
    }
}